File: projects/03/grades.java
/* grades.java is a driver for function grade().
 *
 * Begun by: David Choi, for CS 214 at Calvin College.
 ****************************************************************/

import java.util.Scanner;

public class grades{

   public static void main(String[] args) {
      Scanner keyboard = new Scanner(System.in);
      System.out.println("\nEnter your grade: "); // prompt

      int grade = keyboard.nextInt();       // read grade
      System.out.println(letterGrade(grade)); // display letter grade
   }

   /***************************************************************
    * letterGrade() converts an average grades into letter grades.
    *
    * Receive: grade, an  int.
    * PRE: grade is one of {10, 9, 8, 7, 5, 4, 3, 2, 1}.
    * Return: the letter grade corresponding to average grade.
    */
   public static char letterGrade(int average)
   {
   	switch (average / 10){
   		case 10: case 9:
   		return 'A';

   		case 8:
   		return 'B';

   		case 7:
   		return 'C';

   		case 6:
   		return 'D';

   		default:
   		return 'F';

	}
}
}

File: projects/03/clojure/src/grades.clj
;;;; grades.clj is a driver for function letterGrade.
;;;;
;;;; Input: A grade, a number between 0 to 100.
;;;; Output: The letter for that grade
;;;;  
;;;; Usage: clojure -m grades
;;;;
;;;; Begun by: David Choi, for CS 214 at Calvin College.
;;;; Completed by: David Choi
;;;; Date:02-18-2019
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns grades)                    ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; grades() returns the letter grade for a given grade.
;;; Receive: grade, an Integer.
;;; Precondition: grade is a number between 
;;;   {0 to 100}.
;;; Return: the char corresponding to the grade.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn letterGrade[grade]
	(cond
		(>= grade 90) "A"
		(>= grade 80) "B"
		(>= grade 70) "C"
		(>= grade 60) "D"
		:else "F"
		))
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main() test-drives our letterGrade() function.
;;; Input: grade: one between {0-100}.
;;; Output: the letter grade corresponding to the grade.
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
 (print "\nEnter your grade: ") (flush)
 (let
    [ grade (read) ]                ; read grade (an int)
    (println (letterGrade grade) "\n")      ; display its code
 )
)

